File: src/main/org/nlogo/prim/etc/_self.java
package org.nlogo.prim.etc;

import org.nlogo.agent.Agent;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _self
    extends Reporter {
  @Override
  public Syntax syntax() {
    return Syntax.reporterSyntax
        (Syntax.TYPE_AGENT, "-TPL");
  }

  @Override
  public Object report(final Context context) {
    return context.agent;
  }

  public Agent report_1(final Context context) {
    return context.agent;
  }
}

File: src/main/org/nlogo/prim/etc/_self.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Context, Reporter }

class _self extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(Syntax.AgentType, "-TPL")
  override def report(context: Context) =
    report_1(context)
  def report_1(context: Context) =
    context.agent
}
